internal/controller/stages/stages.go

```diff
@@ -1101,14 +1101,15 @@ func (r *reconciler) syncPromotions(
 	// we order the Promotions from oldest to newest. This is because the
 	// Freight history is garbage collected based on the number of entries,
 	// and we want to ensure that the oldest entries are removed first.
-	slices.SortFunc(newPromotions, func(a, b kargoapi.PromotionInfo) int {
+	slices.SortFunc(newPromotions, func(a, b kargoapi.PromotionReference) int {
 		return strings.Compare(a.Name, b.Name)
 	})
 
 	// Update the Stage status with the information about the newly terminated
 	// Promotions, and any new Freight that was successfully promoted.
 	for _, p := range newPromotions {
 		promo := p
+		promo.CompletionTime = &metav1.Time{Time: time.Now()}
```

Comment:
We can probably be more accurate on the time if we add a `CompletionTime` field to `PromotionStatus` as well and set that when the Promotion actually completes. Then here, we only need to copy it over from the real Promotion to the reference.